src/core/core-async.js

```diff
@@ -14,6 +12,81 @@ let _startRedrawTime = null
 let _groupAll = {}
 let _lastFilteredSize = {}
 
+export class LockTracker {
+  all = false
+  groups = new Set()
+  pendingAll = false
+  pendingGroups = new Set()
+
+  constructor(renderOrRedrawFunc) {
+    this.renderOrRedrawFunc = renderOrRedrawFunc
+  }
+
+  // Utility function to check if a render/redraw should start for the given
+  // group or "all".
+  shouldStart(group, all) {
+    // 1. If currently rendering/redrawing all, return false.
+    // 2. If we're requesting a render/redraw all and *anything* is currently
+    //    rendering/redrawing, return false.
+    // 3. Otherwise, if the requested group is rendering/redrawing, return
+    //    false.
```

Comment:
Is there a usecase here where this returns `true`? 🤔

Reply:
Oh, every case that's not documented to return false =) I suppose I could add a comment about that... done!